### PR TITLE
Added IAM in cartography with documentation and tests

### PR DESCRIPTION
**Pull request by ajinkya1986**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,8 +35,8 @@ You can learn more about the story behind Cartography in our [presentation at BS
 Start [here](docs/setup/install.md).
 
 ## Supported platforms
+- [Google Cloud Platform](docs/setup/config/gcp.md) - Cloud Resource Manager, Compute, DNS, Storage, Google Kubernetes Engine, IAM
```
Comment (mpurusottamc) [inline README.md:38]: Let's keep GCP after AWS line.

File: cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+            "__comment__": "Delete GCP IAM Domains that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:GCPCustomer)-[r:HAS_DOMAIN]->(:GCPDomain) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Remove GCP Customer to IAM Domains relationships that are out of date"
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[:RESOURCE]->(n:GCPDomain) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json:16]: We should not have any reference of CloudanixWorkspace in our feature branches. That node is very specific to our implementation. Please remove all references of CloudanixWorkspace from IAM code.

File: cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Delete GCP IAM Domains that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[r:RESOURCE]->(:GCPDomain) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json:22]: Same here

File: cartography/data/jobs/cleanup/gcp_iam_groups_cleanup.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            "__comment__": "Delete GCP IAM Groups that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:GCPCustomer)-[r:HAS_GROUP]->(:GCPGroups) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Remove GCP Customer to IAM Groups relationships that are out of date"
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[:RESOURCE]->(n:GCPGroup) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_groups_cleanup.json:16]: Same as above

File: cartography/data/jobs/cleanup/gcp_iam_groups_cleanup.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Delete GCP IAM Groups that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[r:RESOURCE]->(:GCPGroup) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_groups_cleanup.json:22]: Same as above

File: cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
+{
+    "statements": [
+        {
+            "query": "MATCH (:GCPCustomer)-[:HAS_DOMAIN]->(n:GCPDomain) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json:4]: All the relationships in cleanup files should be linked to GCPProject. Please update this query to show the relationship of these nodes with GCPProject.

Comment (mpurusottamc) [reply cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json:4]: Also, what does GCPCustomer represent, is it different that GCPUser?

File: cartography/data/jobs/cleanup/gcp_iam_roles_cleanup.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{
+    "statements": [
+        {
+            "query": "MATCH (n:GCPIAMRole)<-[:RESOURCE]-(:GCPProject) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_roles_cleanup.json:4]: For all the cleanup queries, let's also filter GCPProject node to a specific project. See this for reference:
https://github.com/Cloudanix/cartography/pull/36/files#diff-e21ce887de4c4136633b58f57126608fe73076609d7a2ef83597d91fa6960e58R4

File: cartography/data/jobs/cleanup/gcp_iam_roles_cleanup.json [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+            "__comment__": "Delete GCP IAM roles that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:GCPIAMRole)<-[r:RESOURCE]-(:GCPProject) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Remove GCP GCPIAMRole-to-Project relationships that are out of date."
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[:RESOURCE]->(n:GCPRole) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_roles_cleanup.json:16]: Same as above

File: cartography/data/jobs/cleanup/gcp_iam_service_account_keys_cleanup.json [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+            "__comment__": "Delete GCP IAM Service Account Keys that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:GCPProject)-[:RESOURCE]->(:GCPServiceAccount)-[r:HAS_KEY]->(:GCPServiceAccountKey) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Remove GCP Project to IAM Service Account Keys relationships that are out of date"
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[:RESOURCE]->(:GCPServiceAccount)-[:HAS_KEY]->(n:GCPServiceAccountKey) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_service_account_keys_cleanup.json:16]: Same as above

File: cartography/data/jobs/cleanup/gcp_iam_service_accounts_cleanup.json [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+            "__comment__": "Delete GCP IAM Service Account that no longer exist and detach them from all previously connected nodes."
+        },
+        {
+            "query": "MATCH (:GCPIAMServiceAccount)<-[r:RESOURCE]-(:GCPProject) WHERE r.lastupdated <> {UPDATE_TAG} WITH r LIMIT {LIMIT_SIZE} DELETE (r) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100,
+            "__comment__": "Remove GCP GCPIAMServiceAccount-to-Project relationships that are out of date."
+        },
+        {
+            "query": "MATCH (:CloudanixWorkspace)-[:OWNER]->(:GCPProject)-[:RESOURCE]->(n:GCPServiceAccount) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/cleanup/gcp_iam_service_accounts_cleanup.json:16]: Same as above

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
few changes are needed.

File: cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json (hunk 6 above)
Comment (ajinkya1986) [reply cartography/data/jobs/cleanup/gcp_iam_domains_cleanup.json:4]: Domains are connected to only Customers not to projects. Customer is the entity which has a GCP Account.

**Review by mpurusottamc — APPROVED** [review]
looks good. merging.

**Pull request merged**